webpack/assets/javascripts/react_app/redux/actions/notifications/index.js

```diff
@@ -0,0 +1,84 @@
+import {
+  NOTIFICATIONS_GET_NOTIFICATIONS,
+  NOTIFICATIONS_TOGGLE_DRAWER,
+  NOTIFICATIONS_SET_EXPANDED_GROUP,
+  NOTIFICATIONS_MARK_AS_READ
+} from '../../consts';
+import {
+  notificationsDrawer as sessionStorage
+} from '../../../common/sessionStorage';
+import API from '../../../API';
+const getNotificationsInterval = 10000;
+
+export const getNotifications = url => dispatch => {
+  new Promise((resolve, reject) => {
+    if (
+      document.visibilityState === 'visible' ||
+      document.visibilityState === 'prerender'
+    ) {
+      API.get(url).then(response => {
+        dispatch({
+          type: NOTIFICATIONS_GET_NOTIFICATIONS,
+          payload: {
+            notifications: response.notifications
+          }
+        });
+        resolve();
+      }, () => reject());
+    } else {
+      resolve();
+    }
+    return null;
+  }).then(
+    () => {
+      setTimeout(
+        () => dispatch(getNotifications(url)),
+        getNotificationsInterval
+      );
+    },
+    // error handling
+    () => {}
```

Comment:
Would it make sense I think to display some message in the drawer or maybe even change the bell color so the user will have some indication that something failed and what to do about it?